### PR TITLE
unittest and code cleaning

### PR DESCRIPTION
**Pull request by samarthdd**
Updated unit test for all functions
Exception handling
code cleaning according to code standards.
added logger

### CODE REVIEW

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py [hunk 1]
```diff
@@ -1,38 +1,35 @@
...
         settings = get_project_settings()  # Scrapy settings
-
         self.process = CrawlerProcess(settings)
-
-        # print(self.process, )
         self.spidercfg = spidercfg
 
     """ Get all the sites to crawl from config file """
+
     def get_sites_to_run(self, scrape_site=None):
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py:26]: Logging should be done

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
 import os
 import sys
+
 sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
 from scrapy.exceptions import CloseSpider
 from scrapy.crawler import CrawlerProcess
 from scrapy.utils.project import get_project_settings
 from malicious_file_crawler.src.utils.read_config import ConfigReader
 
+logger = logging.getLogger("gw:k8-testdata")
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py:12]: This logger is specific to crawler. So should be name accordingly and stored in separate file.

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py [hunk 3]
```diff
@@ -49,37 +46,33 @@ def get_sites_to_run(self, scrape_site=None):
...
+    if len(sys.argv) == 2:
         scrape_site = sys.argv[1]
         site_list = scraper.get_sites_to_run(scrape_site=scrape_site)
 
     else:
         site_list = scraper.get_sites_to_run()
-    print("in main")
-    scraper.main(site_list)
 
+    logger.info("in main")
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py:77]: This is a candidate for debug logging, not INFO logging.

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/settings.py [hunk 4]
```diff
@@ -74,31 +34,16 @@
 
 DOWNLOAD_TIMEOUT = 12000
 
-FILES_STORE = '../..'
+FILES_STORE = os.getenv("STORAGE_PATH")
 
 # Uncomment this when MINIO service is running
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/settings.py:39]: minio settings should be in separate file. Complete isolation expected.

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/spiders/scraper.py [hunk 5]
```diff
@@ -67,31 +72,41 @@ def errback_handler(self, failure):
...
-            elif ftype == 'text':
-                f.write(data)
-            else:
-                print("Type not recognized.")
+        try:
+            with open(filename, 'w') as f:
+                if ftype == 'json':
+                    json.dump(data, f, indent=2)
+                elif ftype == 'text':
+                    f.write(data)
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/spiders/scraper.py:108]: Use granular exception, to figure out where is the problem and what is the problem. Both read and write operation are under one exception. How you will catch what went wrong.

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/test/test_gw_crawler_runner.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
+from unittest import TestCase, mock
+from scrapy.crawler import CrawlerProcess
+from scrapy.utils.project import get_project_settings
+from malicious_file_crawler.src.glasswall_crawler_runner import GlassWallRunner
+
+
+class TestGlassWallCrawlerRunner(TestCase):
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/test/test_gw_crawler_runner.py:7]: Thanks God. Unit test added. Have you checked code coverage % ?

**Review by susilupwork — APPROVED** [review]
Looks better. Still code need improvement

- All write and Network operation should have INFO logging. 
- Config file and Log file for minio and Scrapper should be separate.
- Use granular exception, to figure out where is the problem and what is the problem. Both read and write operation are under one exception. How you will catch what went wrong. 
- As this is opensource project, Use environment variable for Secrets, keys and credentials. 
- Good you have put unit test cases. Have you checked code coverage % ?

**Review by DinisCruz — APPROVED** [review]
Hi, can you add more details to the Slack channel about the Tech Defence Portal?

**Comment by samarthdd** [comment]
Regarding tekdefence website, It provides sample malware zip files. I have used it for demo and testing since corvus site is no more accessible.

**Pull request merged**